analyze.js

```diff
@@ -104,19 +104,72 @@ async function attachScanFiles(scanGlobs, formData) {
   }
 }
 
+async function validateBranches(branches, config) {
+  const defaultBranch = branches.filter(branch => branch.isDefault)[0].name
+
+  // If target branch is not defined, use the default branch. This also ensures even if
+  // base branch is defined, the default branch is not created as a child of the base branch
+  if (typeof config.targetBranchName == 'undefined') {
+    core.info(`No target branch was defined. Using default target branch '${defaultBranch}'`)
+    config.targetBranchName = defaultBranch
+    config.baseBranchName = undefined
+  }
+
+  const baseBranchPresent = srmClient.checkIfBranchPresent(branches, config.baseBranchName)
+  const targetBranchPresent = srmClient.checkIfBranchPresent(branches, config.targetBranchName)
+
+  if (targetBranchPresent) {
+    core.info("Using existing SRM branch: " + config.targetBranchName);
+    // Not necessary, base branch is currently ignored in the backend if the target
+    // branch already exists. just setting to null to safeguard in case of future changes
+    config.baseBranchName = undefined
+  } else {
+    // Base branch is not defined; throw error
+    if (typeof config.baseBranchName == 'undefined') {
+      throw new Error("A parent branch must be specified when using a new target branch");
+    }
+
+    // Base branch is defined but is not present; cannot create target branch, so throw error
+    if (!baseBranchPresent) {
+      throw new Error("The specified parent branch does not exist: " + config.baseBranchName);
+    }
+
+    // If base branch is defined and present, create the target branch off the base
+    core.info(`Analysis will create a new branch named '${config.targetBranchName}' based on the branch '${config.baseBranchName}'`);
+  }
+}
+
 // most @actions toolkit packages have async methods
 module.exports = async function run() {
   const config = getConfig()
+  const MinimumVersionForBranching = '2022.4.3'
 
-  const client = new CodeDxApiClient(config.serverUrl, config.apiKey, config.caCert)
-  core.info("Checking connection to Code Dx...")
+  const client = new srmClient.SrmApiClient(config.serverUrl, config.apiKey, config.caCert)
+  core.info("Checking connection to SRM...")
 
-  const codedxVersion = await client.testConnection()
-  core.info("Confirmed - using Code Dx " + codedxVersion)
+  const srmVersion = await client.testConnection()
+  core.info("Confirmed - using SRM " + srmVersion)
 
   core.info("Checking API key permissions...")
   await client.validatePermissions(config.projectId)
-  core.info("Connection to Code Dx server is OK.")
+  core.info("Connection to SRM server is OK.")
+
+  core.info("Validating branch selection...")
+  // First check if the SRM version being used supports branching
+  if (srmVersion.localeCompare(MinimumVersionForBranching) < 0) {
```

Comment:
Just a note - this works since `MinimumVersionForBranching` has the same date-based format as the current SRM versions, but this wouldn't work when checking against a minimum version from before the current version naming scheme, e.g. Code Dx version `5.5.0`

To do a proper check, you'd need to split on `.` and compare each element piecewise

Reply:
Updated this, but using the option `{ numeric: true }` supported by localeCompare() to compare numerical strings. This should work as required, ie, `5.5.0` < `2022.3.4` for example.